app/plugins/Wallet/js/lifecycle.js

```diff
@@ -1,89 +1,163 @@
 'use strict';
 
+var blockheight = 0;
+
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Updating  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 // Make API calls, sending a channel name to listen for responses
-function update() {
+function update(address) {
 	IPC.sendToHost('api-call', '/wallet', 'update-status');
-	/*
+
+	// Get list of wallet addresses
 	IPC.sendToHost('api-call', {
-		url: '/wallet/transactions',
+		url: '/wallet/addresses',
+		type: 'GET',
+	}, 'update-address');
+
+	updating = setTimeout(update, 60000 * 5); // update every 5 min
+	return;
+}
+
+// Add transactions to view list per address
+addResultListener('update-address', function(result) {
+	// Update address list
+	eID('address-list').innerHTML = '';
+	result.addresses.forEach( function (address) {
+		appendAddress(address);
+	});
+	return;
+
+	/* Fetch all wallet transactions by iterate over wallet addresses
+	var loopmax = result.addresses.length;
+	var counter = 0;
+	(function next() {
+		setTimeout(function() {
+			updateAddrTxn(result.addresses[counter].address);
+			next();
+		}, 50); // force 50 ms delay between each GET request
+	})();*/
```

Comment:
So, this was a temporary fix to be able to actually view the entire transaction history without the UI freezing. The loop retrieves all transactions *per* address, and then pauses 50 ms before retrieving transactions of the next address. This way, many small datasets are returned and the transaction list in the UI updates in steps. The drawback is that its a real CPU drain and takes a while to execute when you have a large number of addresses. I thus opted for the view-transactions-when-clicking-an-address solution, but left this code in there perhaps as a future option if a user wants to view *all* transactions.